libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1146,6 +1160,65 @@ void WaveClip::Resample(int rate, BasicUI::ProgressDialog *progress)
    }
 }
 
+void WaveClip::ApplyStretchRatio()
+{
+   const auto stretchRatio = GetStretchRatio();
+   if (stretchRatio == 1.0)
+      return;
+
+   Finally Do { [this] {
+      this->mClipStretchRatio = 1.0;
+      this->mRawAudioTempo = this->mProjectTempo;
```

Comment:
I would add `assert(GetStretchRatio() == 1);` which is the postcondition

Reply:
Sure.